contrib/internal/httptrace/httptrace.go

```diff
@@ -90,8 +90,31 @@ func ippref(s string) *netaddr.IPPrefix {
 	return nil
 }
 
+// genClientIPSpanTags generates the client IP related key/value map of tags that need to be added to the span.
+func genClientIPSpanTags(r *http.Request) map[string]string {
+	tags := map[string]string{}
+	ip, matches := getClientIP(r)
+	if matches == nil {
+		if ip.IsValid() {
+			tags[ext.HTTPClientIP] = ip.String()
+		}
+		return tags
+	}
+	sb := strings.Builder{}
+	for hdr, ip := range matches {
+		tags[ext.HTTPRequestHeaders+"."+hdr] = ip
+		sb.WriteString(hdr + ":" + ip + ",")
+	}
+	hdrList := sb.String()
+	tags[ext.MultipleIPHeaders] = hdrList[:len(hdrList)-1]
```

Comment:
This would be a lot safer with a `strings.Join`. So instead, do something like this:

```
        var headers []string
	for header, ip := range matches {
		tags[ext.HTTPRequestHeaders+"."+header] = ip
		headers = append(headers, fmt.Sprintf("%s:%s", header, ip)
	}
	tags[ext.MultipleIPHeaders] = strings.Join(headers, ",")
	return tags
```

Reply:
Great, thanks 👍